File: test/py/myRequest.py
import os, sys
import httplib, urllib
import time
import inspect
import json

class MyRequest:
    def __init__(self, baseUrl, basePort):
        self.baseUrl = baseUrl
        self.basePort = basePort
        
    def stringify(self, payload):
        json_str = json.dumps(payload)
        json_bytes = json_str.encode('utf-8')
        return json_bytes

    def add(self, server_name, load):
        ts=time.time()
        conn = httplib.HTTPConnection(self.baseUrl, self.basePort)
        endpoint = "/servertrack/"+server_name+"/add";
        
        header = {
        	"Content-type": "application/json"
        }
        
        jsonbytes = self.stringify(load)
        
        conn.request("POST", endpoint, jsonbytes, header);
        response = conn.getresponse()
        te=time.time()
        print("XTIME: %s=%dms"%(inspect.currentframe().f_code.co_name, (te-ts)*1000.0))
        if response.status != 200:
            print("ERROR: add failed - endpoint({0}), status({1})".format(endpoint, response.status))
            print("                            message({0}))".format(response.read()))
            return None
        data = response.read()
        print("add:rsp="+data)
        return data

    def getSummary(self, server_name, type):
        ts=time.time()
        conn = httplib.HTTPConnection(self.baseUrl, self.basePort)
        endpoint = "/servertrack/"+server_name+"/report/"+type;
        print("getsummary:"+endpoint)
        
        header = {
            "Content-type": "application/json"
        }
        conn.request("GET", endpoint, None, header);
        response = conn.getresponse()
        te=time.time()
        print("XTIME: %s=%dms"%(inspect.currentframe().f_code.co_name, (te-ts)*1000.0))
        if response.status != 200:
            print("ERROR: getSummary failed - endpoint({0}), status({1})".format(endpoint, response.status))
            print("                            message({0}))".format(response.read()))
            return None
        data = response.read()
        print("getSummary:rsp="+data)
        return data

File: test/js/myRequest.js
var agent = require('request');
var time = require('time');

var method = MyRequest.prototype;

function MyRequest(baseUrl, basePort) {
    this.baseUrl = baseUrl;
    this.basePort = basePort;    
    this.requestUrl = "http://"+this.baseUrl+":"+this.basePort.toString();
    this.header = {
        "Content-Type": "application/json"
    };
}

method.add = function(server_name, load, cb) {
    var endpoint = "/servertrack/"+server_name+"/add"
    var options = {
        url: this.requestUrl+endpoint,
        headers: this.header,
        body: load,
        json: true
    };

    agent.post(options, function(err, response, body) {
        if (err) {
            cb(true, 500);
        }
        else {
        	if (response.statusCode == 200) {
        		//console.log(body);
        		cb(false, body);
        	}
        	else {
        		cb(true, response.statusCode)
        	}
        }
    });
}

method.report = function(server_name, type, cb) {
    var endpoint = "/servertrack/"+server_name+"/report/"+type;
    var options = {
        url: this.requestUrl+endpoint,
        headers: this.header,
        json: true
    };

    agent.get(options, function(err, response, body) {
        if (err) {
            cb(true, 500);
        }
        else {
        	if (response.statusCode == 200) {
        		//console.log(body);
        		cb(false, body);
        	}
        	else {
        		cb(true, response.statusCode)
        	}
        }
    });
}

module.exports = MyRequest;